beaker-vue/src/pages/ChatInterface.vue

```diff
@@ -185,6 +200,47 @@ const isLastCellAwaitingInput = computed(() => {
     return false;
 })
 
+const selectedCellId = computed(() => {
+    return beakerSession.value?.session?.notebook?.selectedCell?.id ?? null;
+});
+
+const selectedCell = computed(() => {
+    if (!selectedCellId.value) return null;
+    const cells = beakerSession.value?.session?.notebook?.cells ?? [];
+    return cells.find(cell => cell.id === selectedCellId.value) || null;
+});
+
+const beakerSession = computed(() => {
+    return beakerInterfaceRef?.value?.beakerSession;
+});
+
+const unselectCell = () => {
+    beakerSession.value.session.notebook.selectedCell = undefined;
+}
+
+const scrollToMessage = () => {
+    const chatCell = document.querySelector(`[data-cell-id="${selectedCellId.value}"]`);
+    if (chatCell) {
+        chatCell.scrollIntoView({ behavior: 'smooth' });
+    }
+}
+
+/**
+ * `selectedCell` indicates that the Thoughts pane should be open,
+ * but clicking the x should always close the right pane.
+ */
+watch(selectedCellId, (newValue) => {
+    if(!rightSideMenuRef.value) return;
+
+    const anyPaneOpen = rightSideMenuRef.value.getSelectedPanelInfo()?.label;
+
+    if (newValue) {
+        rightSideMenuRef.value.selectPanel('Agent Actions');
```

Comment:
Hm.. now that I'm cleaning up- should likely use a constant-ID instead of the label here.

Reply:
Addressed.